faw/pdf-observatory/ui/src/views/HomeView.vue

```diff
@@ -638,6 +647,7 @@ export default Vue.extend({
       decisionSearchCustom: '',
       decisionSearchInsensitive: true,
       decisionSelectedDsl: {} as PdfDecision,
+      extraFeaturesByFile: new Map<string, string[]>(),
```

Comment:
This contains all filenames as strings, and could contain feature text as strings as well, so it's a fairly heavy form in which to store this data versus, say, indices into the file list. We cache it in this form since plugins/search would need it to be converted to this form anyway.

Besides, it's insignificant next to the multiple copies of the decisions we store.

Reply:
May be worth revisiting at some point, but in general, I think you made the right call here regarding avoiding premature optimization.